cmd/access/node_builder/access_node_builder.go

```diff
@@ -736,6 +744,22 @@ func (builder *FlowAccessNodeBuilder) BuildExecutionSyncComponents() *FlowAccess
 				// setup requester to notify indexer when new execution data is received
 				execDataDistributor.AddOnExecutionDataReceivedConsumer(builder.ExecutionIndexer.OnExecutionData)
 
+				// create script execution module, this depends on the indexer being initialized and the
+				// having the register storage bootstrapped
+				scripts, err := execution.NewScripts(
+					builder.Logger,
+					metrics.NewExecutionCollector(builder.Tracer),
+					builder.RootChainID,
+					query.NewProtocolStateWrapper(builder.State),
+					builder.Storage.Headers,
+					execution.IndexRegisterAdapter(builder.ExecutionIndexerCore.RegisterValues),
```

Comment:
I wonder why `ExecutionIndexCore` provides a `RegisterValues` method which get a list of registers?

I think we can change the IndexCore's `RegisterValues` method to `RegisterValue` method, which only work for a single register, this way, we can also get rid of the IndexRegisterAdapter.

The reason I suggest this is that I don't see a use case where we need to get a list of registers at once. If that's the case, we'd better just keep things simple.

Reply:
I think the batch lookups will be needed in the `GetRegisters` endpoint.

Reply:
Yeah, `GetRegisters` is not being used except some tests. I wonder if we can get rid of it?

Reply:
I don't know of a case where we would need a list beside that API. I looked into emulator implementation and it's also fetching a single register at a time. Since FVM always requests a single register I don't know what a usecase is for getting them in batch so I would think changing to a single register is fine, but with this we will break API compatibility. If that's acceptable then we can proceed. Or we could still have a loop that iterates the requested values and gets them.

Reply:
I think we are talking about two things here: 
1. `ExecutionIndexerCore` has a `RegisterValues` which is not being used by anyone
2. `ledger` has a `GetRegisters` which is also not bing used within flow-go codebase, but not sure if any external repo depend on this.


Since `ExecutionIndexerCore` is created by us recently, and its `RegisterValues` is not being used by anyone, I'm suggesting to replace `RegisterValues` with `RegisterValue`, and get rid of the `IndexRegisterAdapter`

Reply:
that sounds reasonable to me. and we can easily add a plural version in the future if needed

Reply:
I think the `GetRegisters` in `ledger` is meant as an abstraction over `[]*ledger.Mtrie`, right? We will still need to implement the plural version for `ExecutionData.GetRegisters`

Reply:
The `GetRegsiters` takes a `state commitment`, which means it's only for a single MTrie
https://github.com/onflow/flow-go/blob/master/storage/ledger.go#L14

Reply:
we are adding a new endpoint to the `ExecutionDataAPI` to get a set of register values (https://github.com/onflow/flow/pull/1389), which will need a method to get a list of registers for a given block height. However, I think that's separate from this discussion, since we could just as easily have 2 methods, `RegisterValue` and `RegisterValues`, or move the loop into the API implementation.

In this case, I think it makes the sense to optimize for the most common case which will be requesting a single register value at a time via script execution.

Reply:
+1 I think it's easier to move the loop to the API

Reply:
I will approve for now. Lets add a RegisterValue to the `ExecutionIndexerCore` in a separate PR, and remove this adapter, since it's adding unnecessary layer.